src/contracts/IbexaTestKernel.php

```diff
@@ -0,0 +1,289 @@
+    }
+
+    public function registerBundles(): iterable
+    {
+        yield new SecurityBundle();
+        yield new IbexaCoreBundle();
+        yield new IbexaLegacySearchEngineBundle();
+        yield new JMSTranslationBundle();
+        yield new FOSJsRoutingBundle();
+        yield new FrameworkBundle();
+        yield new LiipImagineBundle();
+        yield new TwigBundle();
+        yield new DoctrineBundle();
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function registerContainerConfiguration(LoaderInterface $loader): void
+    {
+        $loader->load(static function (ContainerBuilder $container): void {
+            $container->setParameter('ibexa.core.test.resource_dir', self::getResourcesPath());
+        });
+
+        $this->loadConfiguration($loader);
+        $this->loadServices($loader);
+
+        $loader->load(static function (ContainerBuilder $container): void {
+            self::prepareIOServices($container);
+            self::createPublicAliasesForServicesUnderTest($container);
+            self::setUpTestLogger($container);
+        });
+    }
+
+    /**
+     * @throws \Exception
+     */
+    protected function loadConfiguration(LoaderInterface $loader): void
+    {
+        $loader->load(self::getResourcesPath() . '/config/doctrine.php');
+        $loader->load(self::getResourcesPath() . '/config/ezpublish.yaml');
+        $loader->load(self::getResourcesPath() . '/config/framework.yaml');
+        $this->loadSecurity($loader);
+    }
+
+    /**
+     * @throws \Exception
+     */
+    protected function loadServices(LoaderInterface $loader): void
+    {
+        $loader->load(self::getResourcesPath() . '/services/fixture-services.yaml');
+    }
+
+    /**
+     * @throws \Exception
+     */
+    protected function loadSecurity(LoaderInterface $loader): void
+    {
+        $loader->load(self::getResourcesPath() . '/config/security.yaml');
+    }
+
+    /**
+     * @return iterable<class-string>
+     */
+    protected static function getExposedServicesByClass(): iterable
+    {
+        return static::EXPOSED_SERVICES_BY_CLASS;
+    }
+
+    /**
+     * @return iterable<string, class-string>
+     */
+    protected static function getExposedServicesById(): iterable
+    {
+        return static::EXPOSED_SERVICES_BY_ID;
+    }
+
+    private static function getResourcesPath(): string
+    {
+        return __DIR__ . '/Resources';
+    }
+
+    private static function prepareIOServices(ContainerBuilder $container): void
+    {
+        if (!class_exists(InMemoryFilesystemAdapter::class)) {
+            throw new LogicException(
+                sprintf(
+                    'Missing %s class. Ensure that %s package is installed as a dev dependency',
+                    InMemoryFilesystemAdapter::class,
+                    'league/flysystem-memory',
+                )
+            );
+        }
```

Comment:
given this is a contract now, I think you can `composer require` (not dev) `league/flysystem-memory` and drop this check.

Reply:
Actually Flyssytem InMemory adapter was made part of the main package of Flysystem a while ago. There is a high likelihood that this message is no longer accurate.

Reply:
Fixed in 4b868510.